src/main/src/main/java/org/geoserver/catalog/impl/CascadedLayerInfoImpl.java

```diff
@@ -0,0 +1,254 @@
+        return Collections.EMPTY_LIST;
+    }
+
+    @Override
+    public String getPrefferedFormat() {
+        return this.prefferedFormat;
+    }
+
+    @Override
+    public void setPrefferedFormat(String prefferedFormat) {
+        this.prefferedFormat = prefferedFormat;
+    }
+
+    @Override
+    public boolean isSelectedRemoteFormat(String format) {
+        if (prefferedFormat.equalsIgnoreCase(format)) return true;
+        else return selectedRemoteFormats.contains(format);
+    }
+
+    public static boolean isImage(String format) {
+        return format.startsWith("image") && !format.contains("xml") && !format.contains("svg");
+    }
+
+    @Override
+    public Set<StyleInfo> getStyles() {
+        // legacy functionality for empty
+        if (getRemoteStyles().isEmpty()) return super.getStyles();
+
+        return getRemoteStyleInfos()
+                .stream()
+                .filter(s -> !forcedRemoteStyle.equalsIgnoreCase(s.getName()))
+                .filter(s -> selectedRemoteStyles.contains(s.getName()))
+                .collect(Collectors.toSet());
+    }
+
+    @Override
+    public StyleInfo getDefaultStyle() {
+        if (forcedRemoteStyle != null)
+            if (!forcedRemoteStyle.isEmpty()) {
+                Optional<StyleInfo> defaultRemoteStyle =
+                        getRemoteStyleInfos()
+                                .stream()
+                                .filter(s -> s.getName().equalsIgnoreCase(forcedRemoteStyle))
+                                .findFirst();
+
+                if (defaultRemoteStyle.isPresent()) return defaultRemoteStyle.get();
+            }
+        // default to legacy if layer has no forcedRemoteStyle(unlikely)
+        return super.getDefaultStyle();
+    }
+
+    @Override
+    public Set<StyleInfo> getRemoteStyleInfos() {
+        WMSLayerInfo layerInfo = (WMSLayerInfo) super.resource;
+
+        try {
+            return layerInfo
+                    .getWMSLayer(null)
+                    .getStyles()
+                    .stream()
+                    .map(CascadedLayerInfoImpl::getStyleInfo)
+                    .collect(Collectors.toSet());
+
+        } catch (IOException e) {
+            LOGGER.severe(e.getMessage());
+        }
+        // on error default to super
+        return super.getStyles();
+    }
+
+    @Override
+    public Optional<Style> findRemoteStyleByName(final String name) {
+
+        WMSLayerInfo layerInfo = (WMSLayerInfo) super.resource;
+        // temp wrapper for the remote style name
+        final Style style = new NamedStyleImpl();
+        style.setName(name);
+        try {
+            return layerInfo
+                    .getWMSLayer(null)
+                    .getStyles()
+                    .stream()
+                    .filter(s -> s.getName().equalsIgnoreCase(name))
+                    .map(s -> style)
+                    .findFirst();
+        } catch (IOException e) {
+            LOGGER.severe(e.getMessage());
```

Comment:
Same as above, maybe I'm missing something, but if we expect a behavior and it fails then we shoudl fail no? The remote style doesn't exists, how do we recover from that?

Reply:
understood